WORKSPACE

```diff
@@ -14,160 +14,23 @@
-    downloaded_file_path = "futures-3.1.1-py2-none-any.whl",
-    sha256 = "c4884a65654a7c45435063e14ae85280eb1f111d94e542396717ba9828c4337f",
-    # From https://pypi.python.org/pypi/futures
-    urls = [("https://pypi.python.org/packages/a6/1c/" +
-             "72a18c8c7502ee1b38a604a5c5243aa8c2a64f4bba4e6631b1b8972235dd/" +
-             "futures-3.1.1-py2-none-any.whl")],
-)
-
-http_file(
-    name = "futures_2_2_0_whl",
-    downloaded_file_path = "futures-2.2.0-py2.py3-none-any.whl",
-    sha256 = "9fd22b354a4c4755ad8c7d161d93f5026aca4cfe999bd2e53168f14765c02cd6",
-    # From https://pypi.python.org/pypi/futures/2.2.0
-    urls = [("https://pypi.python.org/packages/d7/1d/" +
-             "68874943aa37cf1c483fc61def813188473596043158faa6511c04a038b4/" +
-             "futures-2.2.0-py2.py3-none-any.whl")],
-)
-
-http_file(
-    name = "mock_whl",
-    downloaded_file_path = "mock-2.0.0-py2.py3-none-any.whl",
-    sha256 = "5ce3c71c5545b472da17b72268978914d0252980348636840bd34a00b5cc96c1",
-    # From https://pypi.python.org/pypi/mock
-    urls = [("https://pypi.python.org/packages/e6/35/" +
-             "f187bdf23be87092bd0f1200d43d23076cee4d0dec109f195173fd3ebc79/" +
-             "mock-2.0.0-py2.py3-none-any.whl")],
-)
-
-http_file(
-    name = "google_cloud_language_whl",
-    downloaded_file_path = "google_cloud_language-0.29.0-py2.py3-none-any.whl",
-    sha256 = "a2dd34f0a0ebf5705dcbe34bd41199b1d0a55c4597d38ed045bd183361a561e9",
-    # From https://pypi.python.org/pypi/google-cloud-language
-    urls = [("https://pypi.python.org/packages/6e/86/" +
-             "cae57e4802e72d9e626ee5828ed5a646cf4016b473a4a022f1038dba3460/" +
-             "google_cloud_language-0.29.0-py2.py3-none-any.whl")],
-)
-
-#########################
-# Imports for examples. #
-#########################
-
-pip_import(
-    name = "examples_helloworld",
-    requirements = "//examples/helloworld:requirements.txt",
-)
-
-load(
-    "@examples_helloworld//:requirements.bzl",
-    _helloworld_install = "pip_install",
-)
-
-_helloworld_install()
-
-pip_import(
-    name = "examples_version",
-    requirements = "//examples/version:requirements.txt",
-)
-
-load(
-    "@examples_version//:requirements.bzl",
-    _version_install = "pip_install",
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
+    name = "bazel_federation",
+    url = "https://github.com/bazelbuild/bazel-federation/archive/4da9b5f83ffae17613fa025a0701fa9db9350d41.zip",
+    sha256 = "5b1cf980e327a8f30fc81c00c04007c543e17c09ed612fb645753936de790ed7",
+    strip_prefix = "bazel-federation-4da9b5f83ffae17613fa025a0701fa9db9350d41",
+    type = "zip",
 )
 
-_version_install()
+load("@bazel_federation//:repositories.bzl", "rules_python_deps")
+rules_python_deps()
 
-pip_import(
-    name = "examples_boto",
-    requirements = "//examples/boto:requirements.txt",
-)
+load("@bazel_federation//setup:rules_python.bzl",  "rules_python_setup")
```

Comment:
If rules_python's deps change at head, does that mean I update this function's definition in the federation at head? Or do I override the definition in this WORKSPACE file, and let the federation figure out how to merge my updated requirements into the deps function later on?

Reply:
Yes, you would update deps and setup functions in the federation. Users shouldn't use the federation at head.